File: apps/agor-android/app/src/main/kotlin/live/agor/app/models/Streaming.kt
package live.agor.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Wire shapes for the daemon's streaming events on the messages service.
 * The daemon emits these as `messages streaming:<phase>`, `messages thinking:<phase>`.
 */

@Serializable
data class StreamingStartEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
    @SerialName("task_id") val taskId: String? = null,
    val index: Int? = null,
    val timestamp: String? = null,
)

@Serializable
data class StreamingChunkEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
    val text: String,
    val index: Int? = null,
)

@Serializable
data class StreamingEndEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
    val final: String? = null,
)

@Serializable
data class StreamingErrorEvent(
    @SerialName("session_id") val sessionId: String,
    val error: String,
)

@Serializable
data class ThinkingChunkEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
    val text: String,
)

@Serializable
data class ThinkingStartEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
    @SerialName("task_id") val taskId: String? = null,
    val timestamp: String? = null,
)

@Serializable
data class ThinkingEndEvent(
    @SerialName("session_id") val sessionId: String,
    @SerialName("message_id") val messageId: String? = null,
)

File: apps/agor-rust-android/crates/agor/src/models/streaming.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct StreamingStartEvent {
    pub session_id: String,
    pub message_id: Option<String>,
    pub task_id: Option<String>,
    pub index: Option<i64>,
    pub timestamp: Option<String>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct StreamingChunkEvent {
    pub session_id: String,
    pub message_id: Option<String>,
    pub text: String,
    pub index: Option<i64>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct StreamingEndEvent {
    pub session_id: String,
    pub message_id: Option<String>,
    #[serde(rename = "final")]
    pub final_text: Option<String>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct StreamingErrorEvent {
    pub session_id: String,
    pub error: String,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct ThinkingStartEvent {
    pub session_id: String,
    pub message_id: Option<String>,
    pub task_id: Option<String>,
    pub timestamp: Option<String>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct ThinkingChunkEvent {
    pub session_id: String,
    pub message_id: Option<String>,
    pub text: String,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct ThinkingEndEvent {
    pub session_id: String,
    pub message_id: Option<String>,
}
